extensions/obsidian/src/dailyNoteAppendCommand.tsx

```diff
@@ -19,14 +29,21 @@ interface DailyNoteAppendArgs {
 
 export default function DailyNoteAppend(props: { arguments: DailyNoteAppendArgs }) {
   const { vaults, ready } = useObsidianVaults();
-  const { text } = props.arguments;
   const { appendTemplate, heading, vaultName, silent } = getPreferenceValues<DailyNoteAppendPreferences>();
   const [vaultsWithPlugin, vaultsWithoutPlugin] = vaultPluginCheck(vaults, "obsidian-advanced-uri");
   const [content, setContent] = useState("");
   useEffect(() => {
     async function getContent() {
-      const content = await applyTemplates(text, appendTemplate);
-      setContent(content);
+      try {
+        const selectedText = await getSelectedText();
+        const { text: clipboardText } = await Clipboard.read();
+        const text = props.arguments.text || selectedText || clipboardText;
+        const content = await applyTemplates(text, appendTemplate);
+        setContent(content);
+      } catch (error) {
+        closeMainWindow();
+        console.error("Error in getting content", error);
+      }
```

Comment:
This has the potential to fail for every command if the clipboard fails. Better would be to move the content line outside of the try/catch so it always runs

Also, do you think this might be too surprising of a feature? If the user opens the daily append and presses Enter, I think I would expect it to error, or have the text input pre-filled with the selected/clipboard text, but if it could also then be selected in Raycast so it could be easily deleted (to avoid adding an extra keypress)

What do you think?

Reply:
Excellent suggestion! But I am not sure if displaying error message to users is a good choice cause if the user just wanna input something manually but they got a clipboard fails message, should they feel confusing?

Reply:
I'd have the try/catch fail silently, so have an empty catch

I don't think this feature (pre filling the input) will work though unless the input can be fully selected. Otherwise asking the user to backspace the entire message would be pretty bad.

There's also the case of having lots of text selected or in the history. For example, just copied an entire website text.

Maybe selected text needs to stay separate? 

Do you personally plan to use the clipboard fallback? Seems like the user could just press cmd v right? Maybe it's not that useful?

Reply:
The problem I just want to solve initially is to support selected text as the daily notes input. The clipboard as input feature was a complement by the way. I agree to separate into two commands to be clearer.

Reply:
Maybe just to do the one command then? Seems like a good feature and dont have to worry about side effects